action.yml

```diff
@@ -224,8 +224,20 @@ runs:
           with OUTPUTS_FILE_PATH.open(FILE_APPEND_MODE) as outputs_file:
               outputs_file.writelines(f'{name}={value}{os.linesep}')
 
+      directory = "${{
+        format(
+          '{0}/{1}',
+          (
+            inputs.collection-src-directory
+            && inputs.collection-src-directory
+            || '.tmp-ansible-collection-checkout'
+          ),
+          inputs.collection-root
+        )
+      }}"
+
       COLLECTION_META_FILE = 'galaxy.yml'
-      with open(COLLECTION_META_FILE) as galaxy_yml:
+      with open(os.path.join(directory, COLLECTION_META_FILE)) as galaxy_yml:
```

Comment:
Since this is running on a modern enough Python, maybe it's time to just use `pathlib`?
This all could be as simple as
```python
from pathlib import Path
...
galaxy_yml_path = Path(directory / COLLECTION_META_FILE)
galaxy_yml_text = galaxy_yml_path.read_text()
collection_meta = yaml.loads(galaxy_yml_text)
```